### PR TITLE
Include EventId in telemetry properties

### PR DESCRIPTION
**Pull request by duracellko**
Fix issue #569.

Main update is to `ApplicationInsightsLogger`. It includes EventId and EventName properties in telemetry. It is included only, when EventId has non-default values.
By default this functionality is off for compatibility. Following example can switch it on:
This code should be ASP.NET Core application Startup class.
```C#
public IServiceProvider ConfigureServices(IServiceCollection services)
{
    services.Configure<ApplicationInsightsLoggerOptions>(Configuration.GetSection("ApplicationInsights"));
    services.AddLogging(logging =>
        {
            logging.AddConfiguration(Configuration.GetSection("Logging"))
                .AddConsole()
                .AddDebug();
            logging.Services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider, ApplicationInsightsLoggerProvider>());
        });
}
```

Additionally test coverage of ApplicationInsightsLogger was improved.

### CODE REVIEW

**Comment by cijothomas** [comment]
@duracellko Thanks for your contribution!
We are currently fixing some issues related to symbols, and the next beta release.
Will look at this and merge as soon as the next beta is released.
Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 - Live metrics collection is enabled by default for .NET Core applications (was already enabled for full .NET applications).
 - Updated Web/Base SDK version dependency to 2.5.0-beta1.
 - DependencyCollector referred from 2.5.0-beta1 supports collecting SQL dependency calls in .NET Core Applications using EntityFramework.
+- ApplicationInsightsLogger adds EventId into telemetry properties. It is off by default for compatibility. It can be switched on by configuring ApplicationInsightsLoggerOptions.
```
Comment (cijothomas) [inline CHANGELOG.md]: Can you move this to 2.3.0-beta1 please?

File: src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerProvider.cs [hunk 2]
```diff
@@ -21,10 +25,20 @@ public ApplicationInsightsLoggerProvider(TelemetryClient telemetryClient, Func<s
             this.filter = filter;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationInsightsLoggerProvider"/> class.
+        /// </summary>
+        public ApplicationInsightsLoggerProvider(TelemetryClient telemetryClient, IOptions<ApplicationInsightsLoggerOptions> options)
+        {
+            this.telemetryClient = telemetryClient;
+            this.filter = trueFilter;
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerProvider.cs]: Why use a true filter here?

File: src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerOptions.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.Logging
+{
+    /// <summary>
+    /// Options for configuration of <see cref="ApplicationInsightsLogger"/>.
+    /// </summary>
+    public class ApplicationInsightsLoggerOptions
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerOptions.cs]: There is another PR which introduces ```ApplicationInsightsLoggerOptions```
https://github.com/Microsoft/ApplicationInsights-aspnetcore/pull/574/files
Can you also have took at that as well?

I think passing ```ApplicationInsightsLoggerOptions``` to both ApplicationInsightsLoggerProvider and ApplicationInsightsLogger as done in the other PR is a better option. Can you incorporate this change to your PR?

File: test/Microsoft.ApplicationInsights.AspNetCore.Tests/Logging/ApplicationInsightsLoggerTests.cs [hunk 4]
```diff
@@ -30,6 +30,174 @@ public void TestLoggerSetsSdkVersionOnLoggedTelemetry()
             Assert.True(isCorrectVersion);
         }
 
+        /// <summary>
```
Comment (cijothomas) [inline test/Microsoft.ApplicationInsights.AspNetCore.Tests/Logging/ApplicationInsightsLoggerTests.cs]: Thanks a lot for adding sufficient unit tests!

**Comment by duracellko** [comment]
Thank you for comments. I will have a look at those later this week.

File: src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerOptions.cs (hunk 3 above)
Comment (duracellko) [reply src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerOptions.cs]: Thank you for the hint. Yes, it's actually better. Also I realized that classes are internal, so I can change signature of constructors.

File: src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerProvider.cs (hunk 2 above)
Comment (duracellko) [reply src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerProvider.cs]: I followed pattern in [ConsoleWriter](https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs), but it is not needed with new approach.

**Pull request merged**